crypto/x509/x_crl.c

```diff
@@ -103,13 +103,17 @@ static int crl_set_issuers(X509_CRL *crl)
 
         if (gtmp) {
             gens = gtmp;
-            if (!crl->issuers) {
+            if (crl->issuers == NULL) {
                 crl->issuers = sk_GENERAL_NAMES_new_null();
-                if (!crl->issuers)
+                if (crl->issuers == NULL) {
+                    GENERAL_NAMES_free(gtmp);
                     return 0;
+                }
             }
-            if (!sk_GENERAL_NAMES_push(crl->issuers, gtmp))
+            if (!sk_GENERAL_NAMES_push(crl->issuers, gtmp)) {
+                GENERAL_NAMES_free(gtmp);
                 return 0;
+            }
         }
         rev->issuer = gens;
```

Comment:
```
GENERAL_NAMES_free(rev->issueer);
rev->issuer = gens;
```
??

Reply:
As far as I can tell, `crl->issuers` (which is a private field) is never actually used. It seems to only exist to clean up after `rev->issuer`. So I think an alternative would be to remove `crl->issuers`, give `X509_REVOKED` a callback, nad then make `rev->issuer` an owning pointer.

Reply:
Note that currently more than one `rev->issuer` may share the same pointer since not every iteration creates
a new `gtmp` and therefore reuses the previous `gens`.

Reply:
Oh, it carries over if the extension is missing, which is even what's prescribed in 5280. Wow, yuck.